backend/code/m-booking/src/test/java/com/mbooking/service/UserServiceIntegrationTest.java

```diff
@@ -0,0 +1,245 @@
+package com.mbooking.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.mbooking.dto.EditProfileDTO;
+import com.mbooking.dto.UserDTO;
+import com.mbooking.exception.ApiAuthException;
+import com.mbooking.exception.ApiException;
+import com.mbooking.model.Admin;
+import com.mbooking.model.Authority;
+import com.mbooking.model.Customer;
+import com.mbooking.model.User;
+import com.mbooking.repository.AuthorityRepository;
+import com.mbooking.repository.CustomerRepository;
+import com.mbooking.repository.UserRepository;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test_h2")
+public class UserServiceIntegrationTest {
+
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private UserRepository userRepo;
+
+	@Autowired
+	private CustomerRepository customerRepo;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Autowired
+	private AuthorityRepository authorityRepo;
+
+	@Autowired
+	private EmailSenderService emailSenderService;    
+
+	
+	
+	/*@Before
+	public void setUp() {
+		Customer cus = new Customer();
+		cus.setEmail("email");
+		cus.setEmailConfirmationId("eid");
+		cus.setFirstname("name");
+		cus.setLastname("lastname");
+		cus.setEmailConfirmed(false);
+		cus.isEmailConfirmed();
+
+
+	       Authority roleRegistered = new Authority((long) -2,"ROLE_ADMIN");
+	        authorityRepo.save(roleRegistered);
+
+	        userRepo.save(cus);
+	      
+
+	    }*/
+		
+	
+	
+	
+	
+	@Test(expected = DataIntegrityViolationException.class)
+	public void testRegister() {
```

Comment:
Ok, so we have a couple of issues here:
You created a user dto.
Then you created a customer object.
Then the register method returns a user dto.
In the end, you assert the customer you've created, which will always be the same.